src/vmm/src/linux/vstate.rs

```diff
@@ -579,6 +591,42 @@ impl Vm {
             self.next_mem_slot += 1;
         }
 
+        #[cfg(feature = "amd-sev")]
```

Comment:
IMO we should feature flag this on `#[cfg(feature = "tee")]` instead. Both Mathias and I's PRs are doing this exact same thing. 

In hindsight it may have been better for us to do that as a separate PR to just get the functionality into main knowing all the CC architectures were moving in that direction.

Reply:
Agreed, but I don't think it fits in this PR simply because there is no other TEE arch currently supported. PRs adding TEE arch support should rebase on this and change the `#[cfg(feature = "amd-sev")]` to
`#[cfg(feature = "tee")]`